ignite/engine/__init__.py

```diff
@@ -48,6 +48,7 @@ def supervised_training_step(
     device: Optional[Union[str, torch.device]] = None,
     non_blocking: bool = False,
     prepare_batch: Callable = _prepare_batch,
+    model_transform: Callable[[torch.Tensor], torch.Tensor] = lambda output: output,
```

Comment:
I think the type hint should be a bit more generic: 
```suggestion
    model_transform: Callable[[Any], Any] = lambda output: output,
```